downloadDirectory/src/main/java/org/reactome/release/downloadDirectory/GenerateGOAnnotationFile/GOAGeneratorUtilities.java

```diff
@@ -0,0 +1,247 @@
+     * each type of GO annotation. Depending on if it is looking at the individual protein or whole reaction level, the date attribute
+     * may not be the most recent. If it is found that the goaLine was generated earlier but that a more recent modification date exists based on the
+     * entity that is currently being checked, then it will just update that date value in the hash associated with the line. (Yes, this is weird).
+     * @param entityInst -- GKInstance, Protein/catalyst/reaction that is receiving a GO annotation.
+     * @param goaLine -- String, GO annotation line, used for checking the 'dates' structure.
+     * @return -- int, parsed from the dateTime of the entityInst's modified or created attributes.
+     * @throws Exception -- MySQLAdaptor exception.
+     */
+    public static Integer assignDateForGOALine(GKInstance entityInst, String goaLine) throws Exception {
+        int instanceDate;
+        Collection<GKInstance> modifiedInstances = entityInst.getAttributeValuesList(ReactomeJavaConstants.modified);
+        if (modifiedInstances.size() > 0) {
+            List<GKInstance> modifiedInstancesList = new ArrayList<>(modifiedInstances);
+            GKInstance mostRecentModifiedInst = modifiedInstancesList.get(modifiedInstancesList.size() - 1);
+            instanceDate = getDate(mostRecentModifiedInst);
+        } else {
+            GKInstance createdInst = (GKInstance) entityInst.getAttributeValue(ReactomeJavaConstants.created);
+            instanceDate = getDate(createdInst);
+        }
+
+        // Stores date in global hash that allows date value to be updated if a more recent date was found.
+        if (dates.get(goaLine) == null || instanceDate > dates.get(goaLine)) {
+            dates.put(goaLine, instanceDate);
+        }
+        return instanceDate;
+    }
+
+    /**
+     * Retrieves date from instance and formats it for GO annotation file.
+     * @param instanceEditInst -- GKInstance, instanceEdit from either a Modified or Created instance.
+     * @return -- Integer, from instanceEdit's dateTime. Parsed to remove the Timestamp.
+     * @throws Exception -- MySQLAdaptor exception.
+     */
+    private static int getDate(GKInstance instanceEditInst) throws Exception {
+        return Integer.valueOf(instanceEditInst.getAttributeValue(ReactomeJavaConstants.dateTime).toString().split(" ")[0].replaceAll("-", ""));
+    }
+
+    /**
+     * Iterates through the lines in the 'goaLines' list, retrieves the date associated with that line and also adds the 'Reactome' column before adding it to the gene_association.reactome file.
+     * @throws IOException -- File writing/reading exceptions.
+     */
+    public static void outputGOAFile() throws IOException {
+
+        final String GAF_HEADER = "!gaf-version: 2.1\n";
+
+        Files.deleteIfExists(Paths.get(GOA_FILENAME));
```

Comment:
Could Paths.get(GOA_FILENAME) be consolidated to a temp and re-used in case the value ever needs to be altered?

Reply:
I'm not sure I understand, could you explain a bit further?